_posts/2015-11-16-behat-to-the-fixture.md

```diff
@@ -0,0 +1,272 @@
+         $book1->setTitle('Lord of the ring');
+         $book1->setOwner($userJames); 
+         $manager->persist($book1);
+     
+         $book2 = new Book();
+         $book2->setTitle('Harry Potter 1');
+         $book2->setOwner($userJohn); 
+         $manager->persist($book2);
+     
+         $manager->flush();
+    }
+}
+{% endhighlight %}
+
+
+Or use yml format (with hautelook/alice-bundle) :
+
+{% highlight gherkin %}
+AppBundle\Entity\User:
+    user1: 
+        name: james
+        password: pwd
+    user2:
+        name: john
+        password: pwd
+
+AppBundle\Entity\Book:
+    book1: 
+        title: Lord of the ring
+        owner: @user1
+    book2:
+        title: Harry Potter 1
+        owner: @user2
+{% endhighlight %}
+
+
+## Fixtures in Gherkin language
+
+I propose here a third way, based on Gherkin language and interpreted by behat :
+
+{% highlight gherkin %}
+Feature: Load data.
+
+    Scenario: fixture 1
+        Given users :
+            | name  | password |
+            | james | pwd      |
+            | john  | pwd      |
+        Given books :
+            | title            | user  |
+            | Lord of the ring | james |
+            | Harry Potter 1   | john  |
+
+{% endhighlight %}
+
+Gherkin language is more readable than yml format. Even if you are not a developer, you can write it.
+
+
+## Translate Gherkin language to data
+
+If you are already using behat on your projet, "Given" steps definitions are certainly already defined in one of your Context.
+Example of steps corresponding to fixtures above :
+
+{% highlight php startinline=true %}
+/**
+ * @Given users :
+ */
+public function users(TableNode $tableUsers)
+{
+    foreach ($tableUsers as $userRow) {
+        $user = new User();
+        $user->setName($userRow['name']);
+        $user->setPassword($userRow['password']);
+        $this->em->persist($user);
+    }
+    $this->em->flush();
+}
+
+/**
+ * @Given books :
+ */
+public function books(TableNode $tableBooks)
+{
+    foreach ($tableBooks as $bookRow) {
+        $book = new Book();
+        $book->setTitle($bookRow['title']);
+        $book->setOwner($this->findUserByName($bookRow['owner']));
+        $this->em->persist($book);
+    }
+    $this->em->flush();
+}
+
+{% endhighlight %}
+
+## Behat3 settings
+
+Behat test use test databases, but fixtures have to use dev database. Create *behat_fixtures.yml* file in the root of your project as below :
+
+{% highlight gherkin %}
+default:
+    extensions:
+        Behat\Symfony2Extension:
+            kernel:
+                env: dev
+    suites:
+        fixtures:
+            type: symfony_bundle
+            bundle: 'AppBundle'
+            paths:
+                - src/AppBundle/DataFixtures/
+            contexts :
+                - AppBundle\Features\Context\FixturesContext
+                - ...
+{% endhighlight %}
+
+First you have to define the environment who have to be used : *dev* (default value is *test*) :
```

Comment:
`[...] the environment that should be used [...]`.

Moreover, there is no space between `used` and the colon, nor between the parenthesis and the colon.